awscli/compat.py

```diff
@@ -87,6 +114,8 @@ def compat_open(filename, mode='r', encoding=None):
 
     raw_input = raw_input
 
+    binary_stdin = sys.stdin
```

Comment:
Hmmm we only use this for the codecommit plugin: https://github.com/aws/aws-cli/commit/9f96039f4f24512c2375ff297f4f0d0e76ae8371.

It looks like we do not use it at all for the s3 commands, even for piping to standard out. Do you think we should start using it for both standard in and out? The integration test are not skipped on windows so I feel like we should be fine here as the content is checked for correctness: https://github.com/aws/aws-cli/blob/develop/tests/integration/customizations/s3/test_plugin.py#L1632